### PR TITLE
Enabling build with Visual Studio

### PR DESCRIPTION
**Pull request by natgla**
Enabling build of csmith with VS 2015, both debug and release.
Getting identical code generation from csmith with/without asserts.

### CODE REVIEW

File: src/Block.cpp [hunk 1]
```diff
@@ -426,7 +426,8 @@ Block::append_return_stmt(CGContext& cg_context)
 	ERROR_GUARD(NULL);
 	stms.push_back(sr);
 	fm->makeup_new_var_facts(pre_facts, fm->global_facts);
-	assert(sr->visit_facts(fm->global_facts, cg_context));
+	bool visited = sr->visit_facts(fm->global_facts, cg_context);
+	assert(visited);
```
Comment (regehr) [inline src/Block.cpp]: I don't understand why these changes that turn assert(x) into bool b = x; assert(b); are desirable, could you explain them?

Comment (natgla) [inline src/Block.cpp]: By default release build has no assertions (I changed that for VS in a later commit). In that case this change is needed to produce absolutely same code by release csmith and debug csmith. This is helpful to be able to debug.
I found ~3 such asserts about a year ago; didn't double-check this now.

**Comment by jxyang** [comment]
Yes, function calls in assert should be moved out so they can get evaluated for release build. Except for the safe math change, which I am not sure, all other changes look good to me.

**Comment by regehr** [comment]
The safe math changes seem harmless enough but could you explain the rationale, Natalia? Since it's not obvious it's probably worth explaining that in a comment in the code.

**Comment by natgla** [comment]
Added comment in safe math: 
dnl Non-C99 compliant compilers (e.g. MS VC) may not have hexadecimal floats notation.
dnl For those use ldexp and ldexpf calls instead.

**Comment by regehr** [comment]
Thanks. This LGTM. Could you please squash into a single commit Nadia?

**Comment by natgla** [comment]
Will do this week. Have other urgent work, and already started preparing next series of commits based on this - will need to re-base those.
Natalia.

**Comment by natgla** [comment]
Done. 
Natalia

**Comment by regehr** [comment]
Thanks!
